src/components/DataFetcher/DataFetcher.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+import jobPropTypes from '../Jobs/components/Job/jobPropTypes';
+import {
+  READY_FOR_PREVIEW,
+  PREPARING_FOR_PREVIEW,
+} from '../Jobs/jobStatuses';
+import { DataFetcherContextProvider } from './DataFetcherContext';
+
+const DEFAULT_UPDATE_INTERVAL = 5;
+
+export default class DataFetcher extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.arrayOf(PropTypes.node),
+      PropTypes.node
+    ]).isRequired,
+    mutator: PropTypes.shape({
+      jobsPreviews: PropTypes.shape({
+        GET: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    resources: PropTypes.shape({
+      jobsPreviews: PropTypes.shape({
+        records: PropTypes.arrayOf(
+          PropTypes.shape(
+            PropTypes.arrayOf({
+              jobs: PropTypes.arrayOf(jobPropTypes).isRequired,
+            }).isRequired,
+          ).isRequired,
+        )
+      }),
+    }).isRequired,
+    updateInterval: PropTypes.number, // seconds
+  };
+
+  static defaultProps = {
+    updateInterval: DEFAULT_UPDATE_INTERVAL,
+  };
+
+  static manifest = Object.freeze({
+    jobsPreviews: {
+      type: 'okapi',
+      path: 'metadata-provider/jobExecutions',
+      accumulate: true,
+      throwErrors: false,
+      params: {
+        query: `(status=${READY_FOR_PREVIEW}, ${PREPARING_FOR_PREVIEW})`, // TODO: possible subject to change in future
+      },
+    },
+    // TODO: get required data here
+  });
+
+  state = {
+    contextData: {},
+  }
+
+  componentDidMount() {
+    const { mutator } = this.props;
+    let { updateInterval } = this.props;
+
+    updateInterval *= 1000; // seconds to milliseconds
+
+    // set initial state
+    for (const key of Object.keys(mutator)) {
+      this.setState(({ contextData }) => ({
+        contextData: {
+          ...contextData,
+          [key]: {
+            hasLoaded: false,
+          }
+        }
+      }));
+    }
+
+    this.getData();
+
+    // update data
+    setInterval(this.getData, updateInterval);
```

Comment:
Make sure to cancel interval when the user navigates away from the page. This a kind of memory leak which is bad.